### PR TITLE
Add an action to toggle boolean property values.

### PR DESCRIPTION
**Pull request by D8H**
It avoids the need for a 2nd property to toggle a boolean property value.

### Manual tests
Tested on behavior, shared data and objects.

* https://www.dropbox.com/s/8u83qv80rfal77u/TogglePropertyTest.zip?dl=1

image

### CODE REVIEW

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js [hunk 1]
```diff
@@ -894,9 +900,13 @@ const declarePropertyInstructionAndExpression = (
       entityMetadata.addScopedAction(
         actionName,
         propertyLabel,
-        i18n._(t`Update the property value for ${uncapitalizedLabel}`),
         i18n._(
-          t`Set property value for ${uncapitalizedLabel} of _PARAM0_ to _PARAM${valueParameterIndex}_`
+          t`Update the property value for ${'"' + uncapitalizedLabel + '"'}.`
```
Comment (4ian) [inline newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:904]: Should the quotes be inside the translation? Not all languages use them.

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js [hunk 2]
```diff
@@ -906,6 +916,30 @@ const declarePropertyInstructionAndExpression = (
       .addParameter('yesorno', i18n._(t`New value to set`), '', false)
       .getCodeExtraInformation()
       .setFunctionName(setterName);
+
+    addObjectAndBehaviorParameters(
+      entityMetadata.addScopedAction(
+        toggleActionName,
+        i18n._(t`Toggle ${propertyLabel}`),
+        i18n._(
+          t`Toggle the property value for ${'"' + uncapitalizedLabel + '"'}.`
```
Comment (4ian) [inline newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:925]: Toggle property Xxx. 

Is shorter and better I think.

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js (hunk 1 above)
Comment (D8H) [reply newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:904]: When I tried the translation wasn't working. I don't understand why.

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js (hunk 2 above)
Comment (D8H) [reply newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:925]: Do you mean showing the technical name in the sentence?

Comment (4ian) [inline newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:925]: Sorry for not being clear, I was confused with the other sentence. Here we can have:
```suggestion
          t`Toggle the property value for "${uncapitalizedLabel}".`
```

(quotes are part of the translated string)

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js [hunk 3]
```diff
@@ -906,6 +916,30 @@ const declarePropertyInstructionAndExpression = (
...
+        ) +
+          '\n' +
+          i18n._(
+            `If it was true, it will become false, and if it was false it will become true.`
+          ),
+        i18n._(
+          t`Toggle the property value for ${'"' +
+            uncapitalizedLabel +
+            '"'} of _PARAM0_`
```
Comment (4ian) [inline newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:934]: And here we can do shorter:

```suggestion
          t`Toggle property "${uncapitalizedLabel}" of _PARAM0_`
```

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js (hunk 1 above)
Comment (4ian) [reply newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:904]: Oh that's very strange then

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js [hunk 4]
```diff
@@ -894,9 +900,13 @@ const declarePropertyInstructionAndExpression = (
...
         propertyLabel,
-        i18n._(t`Update the property value for ${uncapitalizedLabel}`),
         i18n._(
-          t`Set property value for ${uncapitalizedLabel} of _PARAM0_ to _PARAM${valueParameterIndex}_`
+          t`Update the property value for ${'"' + uncapitalizedLabel + '"'}.`
+        ),
+        i18n._(
+          t`Set the property value for ${'"' +
+            uncapitalizedLabel +
+            '"'} of _PARAM0_ to _PARAM${valueParameterIndex}_`
```
Comment (4ian) [inline newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js:909]:
```suggestion
          t`Set property "${uncapitalizedLabel}" of _PARAM0_ to _PARAM${valueParameterIndex}_`
```

**Review by 4ian — APPROVED** [review]
See simplification "the property value for Xxx" => "property Xxx".
Would be good to understand why quotes would make the translation fail though.

File: newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js (hunk 1 above)
Comment (D8H) [reply newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js]: Actually, it works. I must have tried to concatenate translatable objects at some point.

**Review by 4ian — APPROVED** [review]

**Pull request merged**